src/Textfield.js

```diff
@@ -60,13 +60,32 @@ class Textfield extends React.Component {
         }
     }
 
+    labelElement(customId) {
+        const { label } = this.props;
+
+        if (typeof label === 'string') {
+            return (
+                <label className="mdl-textfield__label" htmlFor={customId}>
+                    {label}
+                </label>
+            );
+        } else if (typeof label === 'object') {
+            return label;
+        }
+
+        return null;
+    }
+
     render() {
         const { className, inputClassName, id,
               error, expandable, expandableIcon,
               floatingLabel, label, maxRows,
               rows, style, ...otherProps } = this.props;
 
         const hasRows = !!rows;
+        if (!id && typeof label === 'object') {
```

Comment:
I'd prefer having this check in a custom proptype checker function. You can see an example here in the doc: https://facebook.github.io/react/docs/reusable-components.html